src/convert.rs

```diff
@@ -0,0 +1,255 @@
+    let (input_gates, input_wires_num) = {
+        let parts = parse_line(lines.next())?;
+        let input_gates = parts[1..].to_vec();
+        let expected_parties = parts[0];
+        if input_gates.len() != expected_parties {
+            return Err(ConverterError::InputPartiesMismatch(
+                input_gates.len(),
+                expected_parties,
+            ));
+        }
+        let input_wires: usize = input_gates.iter().sum();
+        (input_gates, input_wires)
+    };
+
+    // Parse output line
+    let (mut output_gates, num_output_wires) = {
+        let parts = parse_line(lines.next())?;
+        let num_outputs = parts[0];
+        let gates_per_output = parts[1..].to_vec();
+        if gates_per_output.len() != num_outputs {
+            return Err(ConverterError::OutputCountMismatch(
+                gates_per_output.len(),
+                num_outputs,
+            ));
+        }
+        let total_outputs = gates_per_output.iter().sum();
+        let num_output_wires = gates_per_output.iter().sum::<usize>();
+        (vec![0; total_outputs], num_output_wires)
+    };
+
+    // Create the wires map to map the wires in the Bristol format to the wires in the Garble format.
+    let mut wires_map = vec![0; wires_num];
+    for (i, wire) in wires_map.iter_mut().take(input_wires_num).enumerate() {
+        *wire = i;
+    }
+    let mut next_wire = input_wires_num;
+
+    // Parse gates
+    let mut gates = Vec::new();
+    for line in lines {
+        let parts: Vec<&str> = line.split_whitespace().collect();
+        if parts.len() < 4 {
+            continue;
+        }
+
+        let num_inputs: usize = parts[0]
+            .parse()
+            .map_err(|_| ConverterError::ParseError("Missing number of inputs".to_string()))?;
+        let input_wires: Vec<usize> = parts[2..(2 + num_inputs)]
+            .iter()
+            .map(|s| {
+                s.parse::<usize>()
+                    .map_err(|_| ConverterError::ParseError("Missing input wires".to_string()))
+            })
+            .collect::<Result<_, _>>()?;
+        let output_wire: usize = parts[2 + num_inputs]
+            .parse()
+            .map_err(|_| ConverterError::ParseError("Missing output wires".to_string()))?;
+        let gate_type = parts
+            .last()
+            .ok_or_else(|| ConverterError::ParseError("Missing gate type".to_string()))?;
+
+        // Check if the output wire is an output gate
+        if output_wire < wires_num && output_wire >= wires_num - num_output_wires {
+            output_gates[output_wire - (wires_num - num_output_wires)] = next_wire;
+        }
+
+        wires_map[output_wire] = next_wire;
+        next_wire += 1;
+
+        let gate = match *gate_type {
+            "XOR" => Gate::Xor(wires_map[input_wires[0]], wires_map[input_wires[1]]),
+            "AND" => Gate::And(wires_map[input_wires[0]], wires_map[input_wires[1]]),
+            "INV" => Gate::Not(wires_map[input_wires[0]]),
```

Comment:
I think these three expressions might have potential OOB accesses if the bristol files has lines like
> 1 1 127 0 AND
where the number of specified and provided inputs does not match the operation.